index.js

```diff
@@ -147,10 +147,36 @@ RedirectableRequest.prototype.removeHeader = function (name) {
   this._currentRequest.removeHeader(name);
 };
 
+// Global timeout for all underlying requests
+RedirectableRequest.prototype.setTimeout = function (msecs, callback) {
+  if (callback) {
+    this.once("timeout", callback);
+  }
+
+  var self = this;
+  function startTimer() {
+    clearTimeout(self._timeout);
+    self._timeout = setTimeout(function () {
+      self.emit("timeout");
```

Comment:
So if I was going to abort the request, I'd put that here.  But since apparently the normal `http#setTimeout` doesn't do that, I opted not to.  Let me know what you think.

Reply:
You are correct indeed.